eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala

```diff
@@ -68,6 +70,13 @@ trait Service extends Directives with Logging {
   implicit val actorSystem: ActorSystem
   implicit val mat: ActorMaterializer
 
+  // custom directive to fail with HTTP 404 if the element was not found
+  def complete[T](fut: Future[Option[T]])(implicit marshaller: ToResponseMarshaller[T]): Route = onComplete(fut) {
```

Comment:
If there is no better practice for that, then I think we should define a directive in a separate file and choose a less-ambiguous name.